gp/genome.py

```diff
@@ -53,46 +65,46 @@ def __init__(self, n_inputs, n_outputs, n_columns, n_rows, levels_back, primitiv
             1 + self._primitives.max_arity
         )  # one function gene + multiple input genes
 
-        self._dna = None  # stores dna as list of alleles for all regions
+        self._dna: List[int] = []  # stores dna as list of alleles for all regions
 
         # constants used as identifiers for input and output nodes
         self._id_input_node = -1
         self._id_output_node = -2
-        self._non_coding_allele = None
+        self._non_coding_allele: int = 0
```

Comment:
this is not good as `0` is a valid gene. maybe we can remove this "non-coding allele" completely as suggested in #37

Reply:
Fixed this for now by not initializing this variable.

Reply:
see above regarding uninitialized values